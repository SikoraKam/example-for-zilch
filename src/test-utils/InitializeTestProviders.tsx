import { NavigationContainer } from "@react-navigation/native";
import { FC, PropsWithChildren } from "react";

export const InitializeTestProviders: FC<PropsWithChildren> = ({
  children,
}) => <NavigationContainer>{children}</NavigationContainer>;
