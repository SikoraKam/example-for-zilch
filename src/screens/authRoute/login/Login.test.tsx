import AsyncStorage from "@react-native-async-storage/async-storage";
import { fireEvent, render, waitFor } from "@testing-library/react-native";
import { useForm } from "react-hook-form";
import { InitializeTestProviders } from "../../../test-utils/InitializeTestProviders";
import { Login } from "./Login.container";
import { LoginScreen } from "./Login.screen";

type FormValues = {
  email: string;
  password: string;
};

jest.mock("@react-native-async-storage/async-storage", () => ({
  setItem: jest.fn(),
}));

describe("LoginScreen Test", () => {
  it("renders correctly", () => {
    const TestComponent = () => {
      const { control, handleSubmit, formState } = useForm<FormValues>();
      return (
        <LoginScreen
          control={control}
          handleSubmit={handleSubmit}
          errors={formState.errors}
          onSubmit={jest.fn()}
        />
      );
    };

    const { toJSON } = render(
      <InitializeTestProviders>
        <TestComponent />
      </InitializeTestProviders>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  const LoginContainer = () => (
    <InitializeTestProviders>
      <Login />
    </InitializeTestProviders>
  );
  it("displays validation errors on invalid input", async () => {
    const { getByPlaceholderText, getByText } = render(<LoginContainer />);

    fireEvent.changeText(getByPlaceholderText("E-mail"), "invalidemail");
    fireEvent.changeText(getByPlaceholderText("Password"), "123");
    fireEvent.press(getByText("Log in"));

    await waitFor(() => {
      expect(getByText("email must be a valid email")).toBeTruthy();
      expect(getByText("password must be at least 5 characters")).toBeTruthy();
    });
  });

  it("calls onSubmit with correct data", async () => {
    const { getByPlaceholderText, getByText } = render(<LoginContainer />);

    fireEvent.changeText(getByPlaceholderText("E-mail"), "test@example.com");
    fireEvent.changeText(getByPlaceholderText("Password"), "password123");
    fireEvent.press(getByText("Log in"));

    await waitFor(() => {
      expect(AsyncStorage.setItem).toHaveBeenCalledWith(
        "user",
        JSON.stringify({ username: "John" }),
      );
    });
  });
});
