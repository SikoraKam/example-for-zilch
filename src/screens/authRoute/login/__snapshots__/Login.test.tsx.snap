// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen Test renders correctly 1`] = `
<RNCSafeAreaView
  className="flex-1 bg-background px-4 undefined"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    accessibilityRole="header"
    className="
        flex-row items-center pb-2,
        px-4 sm:px-6,
        justify-center
      "
  >
    <Text
      className="font-normal text-primary font-semibold"
      style={
        {
          "fontSize": 32,
          "lineHeight": 44,
        }
      }
    >
      Log In
    </Text>
  </View>
  <View
    className="mt-12 gap-2"
  >
    <TextInput
      autoCapitalize="none"
      className="py-3 border rounded px-3 undefined"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="E-mail"
    />
    <TextInput
      autoCapitalize="none"
      className="py-3 border rounded px-3 undefined"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
    />
  </View>
  <View
    className="flex-1 justify-end pb-12"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="font-normal text-white text-center undefined"
        style={
          {
            "fontSize": 25,
            "lineHeight": 38,
          }
        }
      >
        Log in
      </Text>
    </View>
  </View>
  <View
    className="pb-3"
  />
</RNCSafeAreaView>
`;
